test/alphabet/nucleotide/nucl16_test.cpp

```diff
@@ -0,0 +1,193 @@
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL KNUT REINERT OR THE FU BERLIN BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ============================================================================
+// Author: Sara Hetzel <sara.hetzel AT fu-berlin.de>
+// ============================================================================
+
+#include <array>
+
+#include <gtest/gtest.h>
+#include <seqan3/alphabet/nucleotide/nucl16.hpp>
+#include <seqan3/alphabet/nucleotide/nucl16_sequence.hpp>
+
+using namespace seqan3;
+
+constexpr char capital_char[] = {'A', 'B', 'C', 'D', 'G', 'H', 'K', 'M', 'N', 'R', 'S', 'T', 'U', 'V', 'W', 'Y'};
+constexpr char lower_char[] = {'a', 'b', 'c', 'd', 'g', 'h', 'k', 'm', 'n', 'r', 's', 't', 'u', 'v', 'w', 'y'};
+
+TEST(alphabet_nucleotides_nucl16_test, default_ctr)
+{
+    constexpr nucl16 nucleotide{};
+
+    EXPECT_EQ(nucleotide, nucl16::A);
+    EXPECT_EQ(nucleotide.value, nucl16::c_type::A);
+}
+
+TEST(alphabet_nucleotides_nucl16_test, copy_ctr)
+{
+    constexpr nucl16 nucleotide1{nucl16::B};
+    constexpr nucl16 nucleotide2{nucleotide1};
+
+    EXPECT_EQ(nucleotide2, nucl16::B);
+    EXPECT_EQ(nucleotide2.value, nucl16::c_type::B);
+}
+
+TEST(alphabet_nucleotides_nucl16_test, move_ctr)
+{
+    constexpr nucl16 nucleotide1{nucl16::C};
+    constexpr nucl16 nucleotide2 = std::move(nucleotide1);
+
+    EXPECT_EQ(nucleotide2, nucl16::C);
+    EXPECT_EQ(nucleotide2.value, nucl16::c_type::C);
+}
+
+TEST(alphabet_nucleotides_nucl16_test, copy_asn)
+{
+    constexpr nucl16 nucleotide1{nucl16::D};
+    nucl16 nucleotide2{nucl16::H};
+
+    nucleotide2 = nucleotide1;
+
+    EXPECT_EQ(nucleotide2, nucl16::D);
+    EXPECT_EQ(nucleotide2.value, nucl16::c_type::D);
+}
+
+TEST(alphabet_nucleotides_nucl16_test, move_asn)
+{
+    constexpr nucl16 nucleotide1{nucl16::K};
+    nucl16 nucleotide2{nucl16::G};
+
+    nucleotide2 = std::move(nucleotide1);
+
+    EXPECT_EQ(nucleotide2, nucl16::K);
+    EXPECT_EQ(nucleotide2.value, nucl16::c_type::K);
+}
+
+TEST(alphabet_nucleotides_nucl16_test, operator_char)
+{
+    constexpr nucl16 nucleotide{nucl16::G};
+    constexpr char c = static_cast<char>(nucleotide);
+
+    EXPECT_EQ(c, 'G');
+}
+
+TEST(alphabet_nucleotides_nucl16_test, to_char)
+{
+    uint8_t rank = 0;
+    for(auto i: capital_char)
```

Comment:
please add a space between for and `(` and also before the `:`

many times in this file :smile: